File: sorting/insertion_sort/insertion_sort.py
def insertion_sort(arr):
    for i in range(1, len(arr)):
        index = i
        while index>0 and arr[index-1] > arr[index]:
            arr[index-1], arr[index] = arr[index], arr[index-1]
            index -= 1

arr = [5,7, 2, 1, 3, 4, 6]
insertion_sort(arr)
print(arr)

File: sorting/insertion_sort/insertion_sort.js
const arr = [6, 2, 8, 3, 5];

const insertion_sort = (arr) => {
  for (let i = 1; i < arr.length; i++) {
    let temp = arr[i];
    let j = i - 1;
    while (j >= 0 && arr[j] > arr[i]) {
      console.log(j);
      arr[j + 1] = arr[j];
      j--;
    }
    arr[j + 1] = temp;
  }
};

function insertionSort(theArray) {
  // For each item in the input array
  for (let i = 1; i < theArray.length; i++) {
    // Shift it to the left until it's in the right spot
    let index = i;
    while (index >= 0 && theArray[index - 1] >= theArray[index]) {
      const tmp = theArray[index - 1];
      theArray[index - 1] = theArray[index];
      theArray[index] = tmp;
      index -= 1;
    }
  }
}

console.log(arr);
insertionSort(arr);
console.log(arr);
